packages/malloy-render/src/component/bar-chart.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files
+ * (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import {LitElement, html} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+import './vega-chart';
+import {DataArray, ExploreField} from '@malloydata/malloy';
+import {consume} from '@lit/context';
+import {resultContext} from './result-context';
+import {RenderResultMetadata} from './render-result-metadata';
+import {getChartSettings} from './chart-settings';
+import {baseSpec} from './vega-lite-base-spec';
+import {valueIsNumber, valueIsString} from './util';
+
+@customElement('malloy-bar-chart')
+export class BarChart extends LitElement {
+  @property({attribute: false})
+  data!: DataArray;
+
+  @consume({context: resultContext})
+  @property({attribute: false})
+  metadata!: RenderResultMetadata;
+
+  override render() {
+    const field = this.data.field as ExploreField;
+    const keys = field.allFields;
+    const {tag} = field.tagParse();
+    const isSpark = tag.text('size') === 'spark';
```

Comment:
NIT: sounds like a good candidate to have some utils for... containsTag?

Reply:
I don't know that we need wrapper utils for the tag APIs themselves, they are pretty robust if you look at what they can do.

But I do find it annoying to have to do `tagParse()` to get the tag, wish I could just call `field.tag.whatever` directly. I'll check with @mtoy-googly-moogly